### PR TITLE
feat(onboarding): lazy onboarding

### LINKED ISSUE #20537: Lazy onboarding PR refresh
### What would you like Renovate to be able to do?

Avoid refreshing onboarding PRs every time there's a commit to the default branch. Onboarding PRs sometime remain open for a long time and users don't need it refreshed non-stop.

### If you have any ideas on how this should be implemented, please tell us here.

For platforms which support checkboxes, only refresh onboarding PRs when either:
- a commit is made to the onboarding branch, or
- users tick a (new) checkbox in the onboarding PR which says something like "Refresh this PR".

If the PR isn't checked or the branch not modified, skip cloning and any other work.

We might need to do some brainstorming on whether the first point requires repositoryCache to be enabled or not. One alternative is to embed the branch commit SHA into the PR body

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Context
 - Closes: https://github.com/renovatebot/renovate/issues/20537

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required
**Need to update the docs**
## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/onboarding/pr/index.ts [hunk 1]
```diff
@@ -215,7 +215,7 @@ async function getRebaseCheckboxComponents(
   }
 
   // Create markdown checkbox
-  rebaseCheckBox = `\n\n---\n\n - [ ] <!-- rebase-check -->If you want to rebase/retry this PR, click this checkbox.\n`;
+  rebaseCheckBox = `\n\n---\n\n - [ ] <!-- rebase-check -->Rebase/refresh this PR from default branch (Warning: custom config changes will be lost).\n`;
```
Comment (rarkins) [inline lib/workers/repository/onboarding/pr/index.ts:218]:
```suggestion
  rebaseCheckBox = `\n\n---\n\n - [ ] <!-- rebase-check -->Rebase/refresh this PR from default branch (Warning: any config changes made will be lost).\n`;
```

File: lib/workers/repository/onboarding/branch/index.spec.ts [hunk 2]
```diff
@@ -314,13 +314,24 @@ describe('workers/repository/onboarding/branch/index', () => {
...
         await checkOnboardingBranch(config);
 
         expect(OnboardingState.prUpdateRequested).toBeFalse();
+        expect(git.checkoutBranch).toHaveBeenCalledTimes(0);
+        expect(git.commitFiles).toHaveBeenCalledTimes(0);
+      });
+
+      it('processes branch if rebase-checkbox in checked', async () => {
+        const pr = { bodyStruct: { rebaseRequested: true } };
+        platform.getBranchPr.mockResolvedValueOnce(mock<Pr>(pr));
```
Comment (viceice) [inline lib/workers/repository/onboarding/branch/index.spec.ts:330]:
```suggestion
        platform.getBranchPr.mockResolvedValueOnce(partial<Pr>(pr));
```
shouldn't we use our helper here?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Comment by RahulGautamSingh** [comment]
Blocked by: https://github.com/renovatebot/renovate/pull/22452

**Pull request closed without merging**